opm/autodiff/BlackoilWellModel_impl.hpp

```diff
@@ -261,16 +280,25 @@ namespace Opm {
     void
     BlackoilWellModel<TypeTag>::
     beginTimeStep() {
+
+        Opm::DeferredLogger local_deferredLogger;
+
         well_state_ = previous_well_state_;
 
         const int reportStepIdx = ebosSimulator_.episodeIndex();
         const double simulationTime = ebosSimulator_.time();
 
-        // test wells
-        wellTesting(reportStepIdx, simulationTime);
+        int exception_thrown = 0;
+        try {
+            // test wells
+            wellTesting(reportStepIdx, simulationTime, local_deferredLogger);
 
-        // create the well container
-        well_container_ = createWellContainer(reportStepIdx);
+            // create the well container
+            well_container_ = createWellContainer(reportStepIdx, local_deferredLogger);
+        } catch (...) {
+            exception_thrown = 1;
+        }
+        OPM_CHECK_FOR_EXCEPTIONS_AND_LOG_AND_THROW(local_deferredLogger, exception_thrown, "BlackoilWellModel::beginTimeStep() failed.", terminal_output_);
```

Comment:
Every call to this macro represents a synchronization point of flow. These should occur very scarcely as the might limit the scalability of it. To there very least there should be quite some amount of computation between them. I doubt that this is the case here.

Reply:
I agree, however, see my remark under.